dart/lib/src/sentry_logger.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+import 'hub.dart';
+import 'hub_adapter.dart';
+import 'protocol/sentry_log.dart';
+import 'protocol/sentry_log_level.dart';
+import 'protocol/sentry_log_attribute.dart';
+import 'sentry_options.dart';
+
+class SentryLogger {
+  SentryLogger(this._clock, {Hub? hub}) : _hub = hub ?? HubAdapter();
+
+  final ClockProvider _clock;
+  final Hub _hub;
+
+  FutureOr<void> trace(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.trace, body, attributes: attributes);
+  }
+
+  FutureOr<void> debug(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.debug, body, attributes: attributes);
+  }
+
+  FutureOr<void> info(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.info, body, attributes: attributes);
+  }
+
+  FutureOr<void> warn(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.warn, body, attributes: attributes);
+  }
+
+  FutureOr<void> error(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.error, body, attributes: attributes);
+  }
+
+  FutureOr<void> fatal(
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    return _captureLog(SentryLogLevel.fatal, body, attributes: attributes);
+  }
+
+  // Helper
+
+  FutureOr<void> _captureLog(
+    SentryLogLevel level,
+    String body, {
+    Map<String, SentryLogAttribute>? attributes,
+  }) {
+    final log = SentryLog(
+      timestamp: _clock(),
+      level: level,
+      body: body,
+      attributes: attributes ?? {},
+    );
+    return _hub.captureLog(log);
+  }
```

Comment:
wonderin when would this ever be not a Future (I'm looking at FutureOr)? 

_hub.captureLog always returns a Future